File: src/WebApp/src/app/models/family.ts
import { Child } from './child';
import { Parent } from './parent';

export interface Family {
    familyId: number;
    familyName: string;
    address: string;
    city: string;
    state: string;
    zip: string;
    telephone: string;
    children?: Child[];
    parents?: Parent[];
}

File: src/Family.API/Models/Family.cs
namespace FamilyApi.Models;

public partial class Family
{
    public int FamilyId { get; set; }

    public string FamilyName { get; set; } = null!;

    public string Address { get; set; } = null!;

    public string City { get; set; } = null!;

    public string State { get; set; } = null!;

    public string Zip { get; set; } = null!;

    public string Telephone { get; set; } = null!;

    public virtual ICollection<Child> Children { get; set; } = new List<Child>();

    public virtual ICollection<Parent> Parents { get; set; } = new List<Parent>();
}
